app/static/js/NeighbourPlotCanvas.js

```diff
@@ -1,6 +1,37 @@
 var _ = require('lodash');
 var d3 = require('d3');
+
 var config = require('../config/plots').neighbourPlot;
+
+// browserify can't load modules dynamically due to the way it statically
+// analyses paths when creating the output bundle. in other words, no variables
+// or concatenated strings can be used in required paths.
+// this switch is inelegant and should be replaced with something nicer
+// maybe this could be configured server-side maybe when matplotlib 1.5 is
+// released and these colour scales become available??
```

Comment:
My idea was to generate the scale (which is just an array of RGB strings)
using matplotlib or Palettable or something on the sever and send that
along with the template, rather than loading up a whole bunch of them in
JS. I'm probably over thinking it though..

On Tue, Feb 16, 2016 at 6:59 PM, Juan Nunez-Iglesias <
notifications@github.com> wrote:

> In app/static/js/NeighbourPlotCanvas.js
> <https://github.com/microscopium/microscopium-ui/pull/78#discussion_r52977637>
> :
>
> >  var config = require('../config/plots').neighbourPlot;
> > +
> > +// browserify can't load modules dynamically due to the way it statically
> > +// analyses paths when creating the output bundle. in other words, no variables
> > +// or concatenated strings can be used in required paths.
> > +// this switch is inelegant and should be replaced with something nicer
> > +// maybe this could be configured server-side maybe when matplotlib 1.5 is
> > +// released and these colour scales become available??
>
> Huh? mpl 1.5 is available both in PyPI and conda. But how does this help
> in js? At any rate I don't find the switch statement *that* offensive!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/microscopium/microscopium-ui/pull/78/files#r52977637>.
>